src/validators/NumberValidator.php

```
@@ -87,26 +104,56 @@ public function init()
+                $this->addError(
+                    $model,
+                    $attribute,
+                    $this->message,
+                );
+
                 return;
             }
+
             $pattern = $this->integerOnly ? $this->integerPattern : $this->numberPattern;
 
             if (!preg_match($pattern, StringHelper::normalizeNumber($value))) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\yii\helpers\StringHelper' in method 'validateAttribute'. (undefined)

(StaticAccess)